pulumi/folio/folio.ts

```diff
@@ -0,0 +1,177 @@
+        appNamespace: k8s.core.v1.Namespace): k8s.core.v1.ConfigMap {
+        return new k8s.core.v1.ConfigMap(name,
+            {
+                metadata: {
+                    name: name,
+                    labels: labels,
+                    namespace: appNamespace.id,
+                },
+                data: data
+            },
+            { provider: cluster.provider });
+    }
+
+    export function secret(name: string, data: any, labels: any, cluster: eks.Cluster,
+        appNamespace: k8s.core.v1.Namespace): k8s.core.v1.Secret {
+        return new k8s.core.v1.Secret(name,
+            {
+                metadata: {
+                    name: name,
+                    labels: labels,
+                    namespace: appNamespace.id,
+                },
+                data: data,
+            },
+            { provider: cluster.provider });
+    }
+
+    /**
+     * Deploy okapi.
+     * @param cluster A reference to the cluster.
+     * @param appNamespace A reference to the app namespace.
+     */
+    export function okapi(module: FolioModule, cluster: eks.Cluster, appNamespace: k8s.core.v1.Namespace) {
+        const values = {
+            // Get the image from the version associated with the release.
+            image: {
+                tag: module.version,
+                repository: `folioorg/${module.name}`
+            },
+
+            // Will handle creating an ingress for this module if true.
+            // ingress: {
+            //     enabled: deployIngress,
+            //     hosts: [
+            //         {
+            //             paths: ["/"]
+            //         }
+            //     ]
+            // },
+
+            service: {
+                type: "LoadBalancer",
+                port: 80,
+                containerPort: 9130
+            },
+
+            // TODO This is known to not always work. So if modules aren't getting registered
+            // with okapi this is likely the problem. The known workaround is to run this script from
+            // this docker image:
+            // https://github.com/folio-org/folio-helm/tree/master/docker/folio-okapi-registration
+            postJob: {
+                enabled: false
+            }
+        }
+
+        deployModuleWithHelmChart(module, cluster, appNamespace, values)
+    }
+
+    /**
+     * Deploys the provided list of folio modules.
+     * @param toDeploy The modules to deploy.
+     */
+    export function modules(toDeploy: Array<FolioModule>,
+        cluster: eks.Cluster,
+        appNamespace: k8s.core.v1.Namespace) {
+        console.log(`Attempting to deploy ${toDeploy.length} modules`);
+
+        for (const module of toDeploy) {
+
+            const values = {
+                // Get the image from the version associated with the release.
+                image: {
+                    tag: module.version,
+                    repository: `folioorg/${module.name}`
```

Comment:
I'm thinking this `modules` method is limited to the the non-special modules for lack of a better term. Anything that isn't okapi, edge and probably stripes. Each of those will probably require their own method since each will be different. See how I did that for `deploy.okapi`.